doc/isis/frr-isis-sonic-config-support-hld.md

```diff
@@ -0,0 +1,174 @@
+
+### Requirements
+
+#### Functional and Configuration Requirements
+
+  1. Extend config and management in SONiC for FRR-ISIS.
+  2. Ability to configre FRR-ISIS based on configurations from config DB with frrcfgd (sonic-buildimage/src/sonic-frr-mgmt-framework) with the field frr_mgmt_framework_config set to "true" in the DEVICE_METADATA table.
+  3. Retain access to FRR UI (vtysh) for managing FRR configs
+  4. Configure FRR-ISIS based on configurations from config DB based on defined template files.
+  5. Upon an FRR container reboot, ISIS template files will restore FRR-ISIS configs read from config DB.
+  6. Provide support for config and management of FRR-ISIS features used in SONIC with new ISIS YANG models.
+
+#### Exemptions
+SONiC CLI support for ISIS show commands are not yet included.
+
+### Architecture Design 
+
+There are no changes to the existing SONiC architecture. This new feature enhances existing code to include configuration support for the isisd daemon within the FRR container. 
+
+### High-Level Design 
+
+#### Design Overview
+
+This feature will extend functionality implemented in [SONiC FRR-BGP Extended Unified Configuration Management Framework](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md) to support additional SONiC FRR-ISIS features. 
+
+The Management framework will convert the YANG-based config data into requests that will write the configs into Redis DB. Redis DB events will trigger frrcfgd when the field frr_mgmt_framework_config set to "true" in the DEVICE_METADATA table, and then frrcfgd will configure FRR-ISIS using FRR CLI commands.
+
+#### Change Overview
+
+This enhancement will support FRR-ISIS features used in SONiC and all changes will reside in the sonic-buildimage repository. Changes include:
+
+- SONiC FRR-ISIS YANG models and YANG validation tests
+  - /src/sonic-yang-models
+- FRR-ISIS config template files and isisd enabled by default in the FRR container
+  - /dockers/docker-fpm-frr
+- Enable ISIS trap messages
+  - /files/image_config/copp
+- Added support for ISIS tables in frrcfgd and extended frrcfgd unit tests for FRR-ISIS configs
+  - /src/sonic-frr-mgmt-framework
+
+#### Container
+
+There will be changes in following containers,
+- Extend frrcfgd support for FRR-ISIS
+  - sonic-mgmt-framework
+- Enable the isisd daemon by default
+  - bgp
+
+### SAI API 
+
+N/A - software feature
+
+### Configuration and management 
+
+#### Manifest
+
+N/A
+
+#### CLI/YANG Model Enhancements
+
+There are no CLI changes made in this feature thus far.
+
+#### Config DB Enhancements  
+
+Following section describes the changes to DB.
+
+Added new configuration tables specific to FRR_ISIS features:
```

Comment:
Would you be adding more details for each of these tables, and also define the schema?

Reply:
doc/isis/frr-isis-sonic-yang-model-hld.md defines these tables